processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3634,7 +3634,7 @@ ExprEval doApply(ExprEval arrayExpr, ExprEval scalarExpr)
       final Object[] array = arrayExpr.asArray();
       final int position = scalarExpr.asInt() - 1;
 
-      if (array.length > position) {
+      if (array.length > position && position >= 0) {
```

Comment:
is this change was coverd with a test? if this is a real issue - I've noticed the same mistake in `ArrayOffsetFunction` :)

Reply:
I didnt add a specific test for this. saw this as an issue & thought of correcting off. Will change in ArrayOffsetFunction as well